src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdClientConfiguration.cs

```diff
@@ -117,14 +117,23 @@ public void Configure(string name, OpenIdConnectOptions options)
         private async Task<OpenIdClientSettings> GetClientSettingsAsync()
         {
             var settings = await _clientService.GetSettingsAsync();
-            if ((await _clientService.ValidateSettingsAsync(settings)).Any(result => result != ValidationResult.Success))
+
+            var result = await _clientService.ValidateSettingsAsync(settings);
+
+            if (result.Any(x => x != ValidationResult.Success))
             {
                 if (_shellSettings.IsRunning())
                 {
-                    _logger.LogWarning("The OpenID Connect module is not correctly configured.");
-                }
+                    if (_logger.IsEnabled(LogLevel.Warning))
+                    {
+                        var errors = result.Where(x => x != ValidationResult.Success)
+                            .Select(x => x.ErrorMessage);
 
-                return null;
+                        _logger.LogWarning("The OpenID client settings are invalid: {Errors}", string.Join(System.Environment.NewLine, errors));
```

Comment:
Let's add a `System` `using` here to remove the `System.` part.

Reply:
Do I need to use the category name because OrchardCore also contains a type with the same name?
``` c#
using SystemEnvironment = System.Environment;

```

Reply:
Ah crap. Yeah, using an alias would work (or you could add the `using` inside the `namespace { ... }`) but it's likely not worth it so let's keep it as-is.